File: 1817/main.cpp
#include <bits/stdc++.h>

using namespace std;
class Solution
{
public:
    vector<int> findingUsersActiveMinutes(vector<vector<int>> &logs, int k)
    {
        unordered_map<int, unordered_set<int>> m;
        for (auto &&log : logs)
        {
            int id = log[0], time = log[1];
            m[id].emplace(time);
        }
        vector<int> ans(k);
        for (auto &[_, minutes] : m)
        {
            int ops = minutes.size();
            ans[ops - 1]++;
        }
        return ans;
    }
};
void solve()
{
    Solution *s = new Solution();
}
int main()
{
    solve();
    return 0;
}

File: 1817/main.swift
class Solution {
    func findingUsersActiveMinutes(_ logs: [[Int]], _ k: Int) -> [Int] {
        var map: [Int: Set<Int>] = [:]
        for log in logs {
            let id = log[0], time = log[1]
            map[id] = (map[id] ?? Set<Int>())
            map[id]!.insert(time)
        }
        var ans = [Int](repeating: 0, count: k)
        for (_, minutes) in map {
            var sz = minutes.count
            ans[sz - 1] += 1
        }
        return ans
    }
}